repobee_sanitizer/_sanitize_repo.py

```diff
@@ -92,6 +97,43 @@ def sanitize_files(
     return []
 
 
+def check_empty_target_branch(
+    repo_path: pathlib.Path, target_branch: str
+) -> bool:
+    repo = git.Repo(str(repo_path))
+    source_branch = repo.head.ref.path
+
+    try:
+        repo.git.symbolic_ref("HEAD", f"refs/heads/{target_branch}")
+        repo.index.diff("HEAD")
+        repo.git.symbolic_ref("HEAD", source_branch)
+    except git.BadName as exc:
+        repo.git.symbolic_ref("HEAD", source_branch)
+        raise EmptyTargetBranchError() from exc
```

Comment:
Reasonably, this function should check that a branch exists. So do that :)

```suggestion
def branch_exists(repo_path: pathlib.Path, branch: str) -> bool:
    repo = git.Repo(str(repo_path))
    existing_branches = {branch.name for branch in repo.branches}
    return branch in existing_branches
```

Also, do we not have type checking running here? Your original function definitely did not return a boolean.

Reply:
FYI you can use a function like this in input validation before starting to run the command. Don't use exceptions for control flow if it can be avoided, it's rather non-obvious, essentially the same as a goto.

I'm not saying don't use exceptions. just don't use them where an `if` can get the job done.